bridgepoint/prebuild.py

```diff
@@ -1609,11 +1615,34 @@ def find_symbol(self, node, name):
             v_var = one(v_int).V_VAR[814]()
         
         return v_var
-    
-    @property
+
+    def get_event_name(self, event):
+        nlevt = one(event).SM_SEVT[525].SM_NLEVT[526]()
+        if nlevt is not None:
+            poly = one(nlevt).SM_PEVT[527]()
+            if poly is not None:
+                if poly.SM_ID == nlevt.SM_ID:  # if this is a local poly
+                    return event.Drv_Lbl + ": " + event.Mning
+                else:
+                    return event.Mning + "::" + poly.localClassName
+            else:
+                return event.Mning + "::Orphaned"
+        else:
+            signalEvt = one(event).SM_SEVT[525].SM_SGEVT[526]()
+            if signalEvt is not None:
+                return event.Drv_Lbl
+            else:
+                return event.Drv_Lbl + ": " + event.Mning
+
+    @ property
     def label(self):
-        return ''
-        
+        action_owner_name = ""
```

Comment:
Is _sm_state always set? If so, this would improve readability:

        action_owner_name = self._sm_state.Name
        if self._sm_evt:
            action_owner_name = self.get_event_name(self._sm_evt)

Reply:
It is not, the TransitionPrebuilder covers the case for SM_ACT for both Transitions and States.